docker-compose.yml

```diff
@@ -19,6 +19,8 @@ services:
     build: './nginx'
     networks:
       main_network:
+    depends_on:
```

Comment:
Why do we need this?

Reply:
As I was testing, I realized that if the nginx container starts before the php container, then it prints a message like `php-docker couldn't be reached` and exits, hence not starting the server!

This seems like an internal dependency that wasn't causing problems because the nginx container tended to start before the php container. I didn't want to mix it with this PR. As this is causing confusion and un-related to the bug fix, I think it might be better to include it in a separate PR actually. I'll remove it from this PR.

Reply:
In the new PR, I will include more background and the error message that comes up if `nginx` starts before `php`.

Reply:
You can add something like https://www.google.co.uk/amp/s/sandro-keil.de/blog/let-nginx-start-if-upstream-host-is-unavailable-or-down/amp/ to the nginx config to make it start even if php hasn't started. Adding `depends on` is brittle (eg what if the php container is unreachable for a second even after it starts)